src/providers/bitcoin/BitcoinSwapProvider.js

```diff
@@ -52,7 +50,16 @@ export default class BitcoinSwapProvider extends Provider {
   }
 
   async claimSwap (initiationTxHash, recipientAddress, refundAddress, secret, expiration) {
-    const secretHash = sha256(secret)
+    return this._redeemSwap(initiationTxHash, recipientAddress, refundAddress, secret, expiration, true)
+  }
+
+  async refundSwap (initiationTxHash, recipientAddress, refundAddress, secretHash, expiration) {
+    return this._redeemSwap(initiationTxHash, recipientAddress, refundAddress, secretHash, expiration, false)
+  }
+
+  async _redeemSwap (initiationTxHash, recipientAddress, refundAddress, secretParam, expiration, isClaim) {
```

Comment:
I have to say the structure of this method has become odd now. It feels like you are not making the right abstractions

For example `isClaim ? recipientAddress : refundAddress` is repeated multiple times. An alternative would be reworking this method to only accept a `to` and pass the result of `isClaim ? recipientAddress : refundAddress` to it instead. 

Similar story for the other params based on `isClaim`

Reply:
I agree. Modified to be more abstract and reuse conditional.